core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/NumberTypeUtils.kt

```diff
@@ -0,0 +1,143 @@
+package org.jetbrains.kotlinx.dataframe.impl
+
+import org.jetbrains.kotlinx.dataframe.impl.api.createConverter
+import org.jetbrains.kotlinx.dataframe.impl.commonNumberType
+import java.math.BigDecimal
+import java.math.BigInteger
+import kotlin.reflect.KClass
+import kotlin.reflect.KType
+import kotlin.reflect.full.withNullability
+import kotlin.reflect.typeOf
+
+/**
+ * Number type graph, structured in terms of number complexity.
+ * A number can always be expressed lossless by a number of a more complex type (any of its parents).
+ *
+ * ```
+ *         BigDecimal
```

Comment:
Ok, found the diagramm here, probably refer to this or just copy-paste to the tests